Document/0x04d-Security-Testing-SDLC.md

```diff
@@ -38,29 +38,44 @@ From the human aspect, this is achieved by creating cross functional teams that
+- Perform a **risk assessment** of the application and its components to identify their risk profile. This risk profile typically depends on the risk appetite of the organization and the regulatory requirements for the application in scope. The risk assessment is additionally influenced by other factors such as whether the application is accessible from the internet, or what kind of data is processed and stored. All sorts of risks need to be taken into account : financial, marketing, industrial, ... A data classification policy determines which data is considered sensitive and prescribes how this data has to be secured.
+- At the beginning of a project or a development cycle, at the same time when functional requirements are gathered, **Security Requirements** are listed and clarified. As use cases are built, **Abuse Cases** are added. Teams (including development teams) may be trained on security if needed (Secure Coding, ...);
+- For mobile applications, the OWASP MASVS (https://www.owasp.org/images/f/fe/MASVS_v0.9.3.pdf) can be leveraged to determine the security requirements based on the risk assessment that was conducted in this initial step. It is common, especially for agile projects, to iteratively review the set of requirements based on newly added features and new classes of data that is handled by the application.
+- Then, as architecture and design are ongoing, a foundational artefact must be performed: **Threat Modeling**, which is b asically an activity where threats are identified, enumerated, prioritized and their treatment initialized. An input of the Threat Model is the **Security Architecture**, but which can be refined after Threat Modeling (both for software and hardware aspects). **Secure Coding rules** are established and the list of **Security tools** that will be used is created. Also, the strategy for **Security testing** is clarified;
+- All security requirements and design considerations should be stored in the Application Life cycle Management System (ALM), which is typically known as issue tracker, that the development/ops team already uses to ensure that security requirements are tightly integrated into the development workflow. The security requirements should ideally also contain the relevant source code snippets for the used programming language, to ensure that developers can quickly reference them. Another strategy for secure coding guidelines is to create a dedicated repository under versioning control, that only contains these code snippets, which has many benefits over the traditional approach of storing these guidelines in word documents or PDFs.
+- The next step is to **develop software**, including security activities such as **Code Reviews** (usually with peers), **Static Analysis** with automated tools and **Unit Tests** dedicated to security;
```

Comment:
In my mind, Security Code Reviews and Static Code Analysis are part of security. Maybe the confusing thing is that they can also be considered as "general" development activities. I would agree with this, however they also have their equivalent in security (hence the "dedicated to security"). Maybe this point need to be clarified, I will propose a different phrasing.